cmd/cluster-node-tuning-operator/main.go

```diff
@@ -2,63 +2,133 @@ package main
 	"github.com/openshift/cluster-node-tuning-operator/pkg/signals"
 	"github.com/openshift/cluster-node-tuning-operator/pkg/tuned"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/util"
 	"github.com/openshift/cluster-node-tuning-operator/version"
 )
 
+const (
+	operandFilename  = "openshift-tuned"
+	operatorFilename = "cluster-node-tuning-operator"
+	metricsHost      = "0.0.0.0"
+)
+
 var (
-	boolVersion = flag.Bool("version", false, "show program version and exit")
-	boolLocal   = flag.Bool("local", false, "local run outside a pod")
+	scheme = apiruntime.NewScheme()
 )
 
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(tunedv1.AddToScheme(scheme))
+	utilruntime.Must(mcov1.AddToScheme(scheme))
+	utilruntime.Must(apiconfigv1.Install(scheme))
+}
+
 func printVersion() {
 	klog.Infof("Go Version: %s", runtime.Version())
 	klog.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
 	klog.Infof("%s Version: %s", tunedv1.TunedClusterOperatorResourceName, version.Version)
 }
 
 func main() {
-	const (
-		operandFilename  string = "openshift-tuned"
-		operatorFilename string = "cluster-node-tuning-operator"
-	)
+	var boolVersion bool
+	var enableLeaderElection bool
+	flag.BoolVar(&boolVersion, "version", false, "show program version and exit")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", true,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
 
 	runAs := filepath.Base(os.Args[0])
 
-	stopCh := signals.SetupSignalHandler()
-
 	switch runAs {
 	case operatorFilename:
+		var metricsAddr string
+		flag.StringVar(&metricsAddr, "metrics-addr", fmt.Sprintf("%s:%d", metricsHost, metrics.MetricsPort),
+			"The address the metric endpoint binds to.")
+
 		klog.InitFlags(nil)
 		flag.Parse()
 
 		printVersion()
 
-		if *boolVersion {
+		if boolVersion {
 			os.Exit(0)
 		}
 
-		go metrics.RunServer(metrics.MetricsPort, stopCh)
-		metrics.RegisterVersion(version.Version)
-		controller, err := operator.NewController()
+		// We have two namespaces that we need to watch:
+		// 1. NTO namespace - for NTO resources
+		// 2. None namespace - for cluster wide resources
+		ntoNamespace := config.OperatorNamespace()
+		namespaces := []string{
+			ntoNamespace,
+			metav1.NamespaceNone,
+		}
+
+		restConfig := ctrl.GetConfigOrDie()
+		le := util.GetLeaderElectionConfig(restConfig, enableLeaderElection)
+		mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+			NewCache:                cache.MultiNamespacedCacheBuilder(namespaces),
+			Scheme:                  scheme,
+			LeaderElection:          true,
+			LeaderElectionID:        config.OperatorLockName,
+			LeaderElectionNamespace: ntoNamespace,
+			LeaseDuration:           &le.LeaseDuration.Duration,
+			RetryPeriod:             &le.RetryPeriod.Duration,
+			RenewDeadline:           &le.RenewDeadline.Duration,
+			Namespace:               ntoNamespace,
```

Comment:
You are passing an `NS and NewCache: cache.MultiNamespacedCacheBuilder(namespaces)`
Note that you need to have permissions ( scope ) to read/update/delete resources and in this way cache them where the operator will be installed. Then, you can:

a) [Watching/Catching resources in a set of Namespaces](https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/#watching-resources-in-a-set-of-namespaces)
It is possible to use [MultiNamespacedCacheBuilder](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) from [Options](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#Options) to watch and manage resources in a set of Namespaces.

OR 

b) Watching/Catching resources in a single Namespace (where the. operator will be installed) by using the Namespace option. See: 

> // Namespace if specified restricts the manager's cache to watch objects in
> // the desired namespace Defaults to all namespaces
> //
> // Note: If a namespace is specified, controllers can still Watch for a
> // cluster-scoped resource (e.g Node).  For namespaced resources the cache
> // will only hold objects from the desired namespace.
> Namespace [string](https://pkg.go.dev/builtin#string)
 
Ref: https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.1/pkg/manager#Manager

OR 

c) Do not add both which means grant cluster-scope permission for the project. Your operator will be watching/catching the whole cluster 

Be aware that how much more resources/namespaces do you catching/watching more resources you will consume.